net/relayd/src/opnsense/mvc/app/controllers/OPNsense/Relayd/Api/StatusController.php

```diff
@@ -68,47 +69,99 @@ public function sumAction($wait=0)
         if (empty($output[0])) {
             return $result;
         }
+        $output[] = "0\t****\t"; // end of data marker
         $result["result"] = 'ok';
         $virtualServerId = 0;
         $virtualServerType = '';
         $tableId = 0;
         $virtualserver = array();
         $rows = array();
         foreach ($output as $line) {
-            $words = explode("\t", $line);
-            $id = trim($words[0]);
-            $type = trim($words[1]);
-            if ($type == 'redirect' || $type == 'relay') {
+            $words = array_map('trim', explode("\t", $line));
+            $id = $words[0];
+            $type = $words[1];
+            if ($type == 'redirect' || $type == 'relay' || $type == '****') {
                 // new virtual server id/type means new record
                 if (
-                    ($id != $virtualServerId
-                    && $virtualServerId > 0)
-                    || ($type != $virtualServerType
-                    && strlen($virtualServerType) > 5)
+                    ($id != $virtualServerId && $virtualServerId > 0) ||
+                    ($type != $virtualServerType && strlen($virtualServerType) > 5) ||
+                    ($type == '****' && !empty($virtualserver))
                 ) {
+                    // append backend hosts not found in the list, since relayd only supports disabled tables
+                    // you might loose track of hosts that are disabled
+                    if (!empty($virtualserver['tables'])) {
+                        foreach ($virtualserver['tables'] as &$table) {
+                            if (!empty($table['uuid'])) {
+                                $tblnode = $relaydMdl->getNodeByReference("table.".$table['uuid']);
+                                foreach (explode(",", (string)$tblnode->hosts) as $host_uuid) {
+                                    $found = false;
+                                    if (!empty($table['hosts'])) {
+                                        foreach ($table['hosts'] as $tblhost) {
+                                            if (isset($tblhost['uuid']) && $tblhost['uuid'] == $host_uuid) {
+                                                $found = true;
+                                            }
+                                        }
+                                    } else {
+                                        $table['hosts'] = [];
+                                    }
+                                    if (!$found) {
+                                        $hostnode = $relaydMdl->getNodeByReference("host.".$host_uuid);
+                                        $table['hosts'][$host_uuid] = [
+                                            "name" => (string)$hostnode->name,
```

Comment:
Relayd uses the address not the name
must be `"name" => (string)$hostnode->address,`

Reply:
@fbrendel oops, thanks, I'll put it on my list to change tomorrow.

Reply:
@fbrendel do you know if there's a use case to add the same host address multiple times? would that act similar to a weight criteria on multi wan scenario's?

Reply:
The same address to multiple hosts? From the Relayd point of view no.
If you have redirects the backends must serve the virtual IP. So yes the have the same (local) assigned.

But for Relayd you don't need that. You can assign a host to multiple tables.

From the plugin point of view it could make an organizational sense.
E.g. it could be easier when you add a new backend. You don't need to create a new host and add it to the dozens of corresponding tables.

Reply:
I think we can serve the organisational part when matching the hosts that are linked to a table, I've just added a commit which does that. Let's try to fluff the status page a bit and see what it brings.